### PR TITLE
Fire map events for touchstart, touchend, touchmove, touchcancel

### LINKED ISSUE #2195: Touch events don't trigger mapbox events
### Context

* **mapbox-gl-js version**:  v0.14.3
* **devices**:  Nexus 5, iPhone 6s, MBP chrome device emulator

### Report

**Expected behavior**: Tapping map fires 'click' event. Same for doubletap: 'doubleclick',  touchstart: 'mousedown', touchend: 'mouseup'

**Actual behavior**: No events fired

**Steps to reproduce the behavior**:

1. Instantiate gl map
2. Add 'click' event listener to map
3. Touch map

### PR DESCRIPTION
**Pull request by bhousel**
(closes #2195)

### CODE REVIEW

File: js/util/browser/dom.js [hunk 1]
```diff
@@ -56,5 +56,18 @@ exports.mousePos = function (el, e) {
...
+exports.touchPos = function (el, e) {
+    var rect = el.getBoundingClientRect(),
+        points = [];
+    for (var i = 0; i < e.touches.length; i++) {
+        points.push(new Point(
+            e.touches[i].clientX - rect.left - el.clientLeft,
+            e.touches[i].clientY - rect.top - el.clientTop
+        ));
+    }
+    return points;
```
Comment (lucaswoj) [inline js/util/browser/dom.js]: What do you think about inlining this method into `_fireTouchEvent`? When might we reuse this code?

File: js/util/browser/dom.js [hunk 2]
```diff
@@ -56,5 +56,18 @@ exports.mousePos = function (el, e) {
...
+
+exports.touchPos = function (el, e) {
+    var rect = el.getBoundingClientRect(),
+        points = [];
+    for (var i = 0; i < e.touches.length; i++) {
+        points.push(new Point(
+            e.touches[i].clientX - rect.left - el.clientLeft,
+            e.touches[i].clientY - rect.top - el.clientTop
+        ));
+    }
```
Comment (lucaswoj) [inline js/util/browser/dom.js]: You could simplify this code using `Array#map`.

**Comment by lucaswoj** [comment]
This is solid code @bhousel :bow:

I left some minor suggestions. Merge this PR at your discretion. :ship:

File: js/util/browser/dom.js (hunk 2 above)
Comment (bhousel) [inline js/util/browser/dom.js]: Nope!   [`TouchList`](https://developer.mozilla.org/en-US/docs/Web/API/TouchList) is one of those special DOM collections (like `NodeList`, `ClassList`, etc) that doesn't include all of the useful functions from `Array.prototype` that one would expect them to include.

File: js/util/browser/dom.js (hunk 1 above)
Comment (bhousel) [inline js/util/browser/dom.js]: I'm not sure, but maybe anywhere that we use `DOM.mousePos` (like the boxzoom or touchpanrotate code) we might change to possibly support dispatching an event with multitouch properties.

File: js/util/browser/dom.js (hunk 2 above)
Comment (lucaswoj) [inline js/util/browser/dom.js]: `Array.prototype.map.call(e.touches, function(touch) { ... })` :smiling_imp: 

Either way is ok :+1:

File: js/util/browser/dom.js (hunk 1 above)
Comment (lucaswoj) [inline js/util/browser/dom.js]: Ah! I didn't know that this was a foil to another method called `mousePos`. I'm a sucker for consistency.

**Comment by lucaswoj** [comment]
Looks great! :shipit:

**Comment by bhousel** [comment]
merged!

**Pull request closed without merging**